js/exporter/jspdf/rows_generator.js

```diff
@@ -0,0 +1,98 @@
+import { isDefined } from '../../core/utils/type';
+
+// Returns IPdfRowInfo[]
+// [
+//    {
+//      rowType, - readonly
+//      rowIndex, - readonly
+//      indentLevel, - readonly
+//      cells: [ - readonly
+//        {
+//          gridCell, - readonly
+//          colSpan, - readonly (for internal use/hide from api/useless???)
+//          rowSpan, - readonly (for internal use/hide from api/useless???)
+//          text,
+//          wordWrapEnabled,
+//        }
+//      ],
+//    }
+// ]
+
+function generateRowsInfo({ dataProvider, dataGrid, headerStyles }) {
```

Comment:
This method is called with arguments as list, not a single object.
https://github.com/DevExpress/DevExtreme/pull/19048/files#diff-2b4d5ffc08be59730b08b9c76631266e6eb7281862fa44e2465cb21e6aa68c75R25

```
const rowsInfo = generateRowsInfo(dataProvider, dataGrid);
```